pkg/storage/chunk/client/alibaba/oss_object_client.go

```diff
@@ -0,0 +1,158 @@
+func (s *OssObjectClient) GetObject(ctx context.Context, objectKey string) (io.ReadCloser, int64, error) {
+	var resp *oss.GetObjectResult
+	var options []oss.Option
+	err := instrument.CollectedRequest(ctx, "OSS.DeleteObject", ossRequestDuration, instrument.ErrorCode, func(ctx context.Context) error {
+		var requestErr error
+		resp, requestErr = s.defaultBucket.DoGetObject(&oss.GetObjectRequest{ObjectKey: objectKey}, options)
+		if requestErr != nil {
+			return requestErr
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, 0, err
+	}
+	length := resp.Response.Headers.Get("Content-Length")
+	size, err := strconv.Atoi(length)
+	if err != nil {
+		return nil, 0, err
+	}
+	return resp.Response.Body, int64(size), err
+
+}
+
+// PutObject puts the specified bytes into the configured OSS bucket at the provided key
+func (s *OssObjectClient) PutObject(ctx context.Context, objectKey string, object io.ReadSeeker) error {
+	return instrument.CollectedRequest(ctx, "OSS.PutObject", ossRequestDuration, instrument.ErrorCode, func(ctx context.Context) error {
+		if err := s.defaultBucket.PutObject(objectKey, object); err != nil {
+			return errors.Wrap(err, "failed to put oss object")
+		}
+		return nil
+	})
+
+}
+
+// List implements chunk.ObjectClient.
+func (s *OssObjectClient) List(ctx context.Context, prefix, delimiter string) ([]client.StorageObject, []client.StorageCommonPrefix, error) {
+	var storageObjects []client.StorageObject
+	var commonPrefixes []client.StorageCommonPrefix
+	marker := oss.Marker("")
+	for {
+		if ctx.Err() != nil {
+			return nil, nil, ctx.Err()
+		}
+
+		objects, err := s.defaultBucket.ListObjects(oss.Prefix(prefix), oss.Delimiter(delimiter), marker)
+		if err != nil {
+			return nil, nil, errors.Wrap(err, "list alibaba oss bucket failed")
+		}
+		marker = oss.Marker(objects.NextMarker)
+		for _, object := range objects.Objects {
+			storageObjects = append(storageObjects, client.StorageObject{
+				Key:        object.Key,
+				ModifiedAt: object.LastModified,
+			})
+		}
+		for _, object := range objects.CommonPrefixes {
+			if object != "" {
+				commonPrefixes = append(commonPrefixes, client.StorageCommonPrefix(object))
+			}
+		}
+		if !objects.IsTruncated {
+			break
+		}
+	}
+	return storageObjects, commonPrefixes, nil
+}
+
+// DeleteObject deletes the specified object key from the configured OSS bucket.
+func (s *OssObjectClient) DeleteObject(ctx context.Context, objectKey string) error {
+	return instrument.CollectedRequest(ctx, "OSS.DeleteObject", ossRequestDuration, instrument.ErrorCode, func(ctx context.Context) error {
+		err := s.defaultBucket.DeleteObject(objectKey)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+}
+
+// IsObjectNotFoundErr returns true if error means that object is not found. Relevant to GetObject and DeleteObject operations.
+func (s *OssObjectClient) IsObjectNotFoundErr(err error) bool {
+	return errors.Is(err, storage.ErrObjectNotExist)
```

Comment:
This looks copied/pasted again and is comparing google errors to alibaba responses

Reply:
thanks.
The IsObjectNotFoundErr return value of error has been checked in detail.
At the same time, the test function has been added to confirm that it is running correctly.

ossErrorDeatil message snapshot.
![image](https://user-images.githubusercontent.com/9583245/217410651-fb437883-f29f-4553-b623-0874a3699785.png)
test case pass. IsObjectNotFoundErr and GetObject
![image](https://user-images.githubusercontent.com/9583245/217412362-b6fe30ad-44c5-4fa2-ba21-0eb0df7576de.png)